Partially update orders shipments etc

Sin0fGREED opened the pull request:
Deze pull request introduceert verschillende nieuwe eindpunten voor gedeeltelijke updates in meerdere controllers en bevat wat code-refactoring in de testbestanden voor betere leesbaarheid en consistentie. De belangrijkste wijzigingen omvatten het toevoegen van nieuwe `HttpPatch` eindpunten aan verschillende controllers en het refactoren van testbestanden voor verbeterde opmaak.

### Nieuwe `HttpPatch` Eindpunten:
* [`C#api/Controllers/ClientsController.cs`](diffhunk://#diff-1184672ff44a5aadf8f1006a515b0e9af4d030315f4cd512401e7d884d1a246eR77-R91): Toegevoegd `PartiallyUpdateClient` eindpunt om gedeeltelijke updates voor klanten te verwerken.
* [`C#api/Controllers/InventoriesController.cs`](diffhunk://#diff-39fa7fea249804cc49a931e2bb5006bc5207338e33b5ec6ffcba782875efb589R67-R81): Toegevoegd `PartialUpdateInventory` eindpunt om gedeeltelijke updates voor inventarissen te verwerken.
* [`C#api/Controllers/Item_GroupsController.cs`](diffhunk://#diff-ad02622340c889888c14778ee63f90716659e62c2829149bc5fe275b0db026dcR77-R91): Toegevoegd `PartialUpdateItemGroup` eindpunt om gedeeltelijke updates voor itemgroepen te verwerken.
* [`C#api/Controllers/Item_LinesController.cs`](diffhunk://#diff-55389e6d43f65e50c19333c4f844a4d85bcfc2d7f7dda528c10ec44c5d7610d4R77-R91): Toegevoegd `PartialUpdateItemLine` eindpunt om gedeeltelijke updates voor itemregels te verwerken.
* [`C#api/Controllers/Item_TypesController.cs`](diffhunk://#diff-013af184419daa183e49908066ff38c40cc9c1925b09e6dbd635d2aeb82d9483R77-R91): Toegevoegd `PartialUpdateItemType` eindpunt om gedeeltelijke updates voor itemtypen te verwerken.
* [`C#api/Controllers/ItemsController.cs`](diffhunk://#diff-9448455ae727a34536bbcbac5a53961a68796e35c74145f1121778abfea60bbeR100-R114): Toegevoegd `PartialUpdateItems` eindpunt om gedeeltelijke updates voor items te verwerken.
* [`C#api/Controllers/ShipmentsController.cs`](diffhunk://#diff-36914a8e7531c78f1a4e91bba1a56d1f5e3bdf297795b2fe409b9da1614cddeeR99-R113): Toegevoegd `PartialUpdateShipment` eindpunt om gedeeltelijke updates voor zendingen te verwerken.
* [`C#api/Controllers/SuppliersController.cs`](diffhunk://#diff-477fda09061924b307ab77ebc3a7e4ef4f090d0daecc1d0063520e1f6f429181R77-R91): Toegevoegd `PartialUpdateSupplier` eindpunt om gedeeltelijke updates voor leveranciers te verwerken.
* [`C#api/Controllers/TransfersController.cs`](diffhunk://#diff-c46f0164c9500d1cf066d559e1ce644cc6033cfcae3327891b68cca57c79b5ecR89-R104): Toegevoegd `PartialUpdateTransfer` eindpunt om gedeeltelijke updates voor overgangen te verwerken.

### Refactoring van Testbestanden:
* [`C#api/Tests/test_clients.py`](diffhunk://#diff-06a20c39515e9ee3e299bd3e83a8f50ce5a103227908fc31df4ef34609300b1dL1-R22): Verbeterde importstatements en herschikte code voor betere leesbaarheid. [[1]](diffhunk://#diff-06a20c39515e9ee3e299bd3e83a8f50ce5a103227908fc31df4ef34609300b1dL1-R22) [[2]](diffhunk://#diff-06a20c39515e9ee3e299bd3e83a8f50ce5a103227908fc31df4ef34609300b1dL30-R72) [[3]](diffhunk://#diff-06a20c39515e9ee3e299bd3e83a8f50ce5a103227908fc31df4ef34609300b1dL102-R178)
* [`C#api/Tests/test_inventories.py`](diffhunk://#diff-84e3498cf2c6213beb2f1a63d5fc1301c07c1c90ff4deaa43645bc0a426ff52cL1-R22): Verbeterde importstatements en herschikte code voor betere leesbaarheid. [[1]](diffhunk://#diff-84e3498cf2c6213beb2f1a63d5fc1301c07c1c90ff4deaa43645bc0a426ff52cL1-R22) [[2]](diffhunk://#diff-84e3498cf2c6213beb2f1a63d5fc1301c07c1c90ff4deaa43645bc0a426ff52cL29-R54) [[3]](diffhunk://#diff-84e3498cf2c6213beb2f1a63d5fc1301c07c1c90ff4deaa43645bc0a426ff52cL52-R65)

Sin0fGREED committed:
Partially update added

Testing part is still left. but it should work otherwise i can change the "var success = DataProvider.fetch_client_pool().UpdateClient(id, partialClient);"
in to "var success = DataProvider.fetch_client_pool().UpdateClient(id, partialClient);var success = DataProvider.fetch_client_pool().PartialUpdateClient(id, partialClient);"
and add a method.

Sin0fGREED committed:
Fixed flake8

Sin0fGREED committed:
Tests verbeterd

safouuwa commented:
Functionaliteit is nog niet werkend zoals verwacht.

![image](https://github.com/user-attachments/assets/23b33fb2-0301-478a-b4fa-efcaf8e145cc)
Dit geval zou goed gekeurd moeten worden volgens de user story, de user wilt alleen de name field van client met id 1 veranderen, maar het programma verwacht toch alle andere fields erbij.

![image](https://github.com/user-attachments/assets/0ac469e2-bca4-4b2d-b4ab-304ba8341807)
Daarnaast zijn de tests ook niet geslaagd (ook de test job in de workflow run geeft dit aan)

![image](https://github.com/user-attachments/assets/74b1f0b9-e3ce-467a-971f-c3c23ff16d8c)
Ook door sommige (onnodige) wijzingen in de tests die fundamenteel geen verschil op leveren.

Sin0fGREED committed:
Adding replace function

Fix 0.5, Instead of having to change all of the fields you can instead change only one when editing partially

Sin0fGREED committed:
added Replace itemLines

Sin0fGREED committed:
Updates

Sin0fGREED committed:
Updated clients

Sin0fGREED committed:
small changes

The patch in thunderclient shows a response but it doesn't get replaced

Sin0fGREED committed:
Fixed the patch in clients.cs and its corresponding controller.

Sin0fGREED committed:
final fix for clients.

Sin0fGREED committed:
test clients

Sin0fGREED committed:
Merge branch 'merge-branch-devvelopment' into partially-update-orders-shipments-etc

Sin0fGREED committed:
reverted a couple changes in test to a working version and commented patch

Sin0fGREED committed:
Update InventoriesController.cs

Sin0fGREED committed:
added itemGroup patch

Sin0fGREED committed:
added item line patch

Sin0fGREED committed:
added item type patch and fixed typo in some patches

Sin0fGREED committed:
Added item.cs and made the patch code more identical to each other

Sin0fGREED committed:
added Shipment patch method

Sin0fGREED committed:
added suppliers patch method

Sin0fGREED committed:
Added Transfers patch method

Sin0fGREED committed:
added patch warehouse method

Sin0fGREED committed:
added location patch method and added timestamp in warehouse

Sin0fGREED committed:
added order patch Method

Sin0fGREED committed:
test Patch method

linkverk commented:
Ik zie dat je veel hebt gefixed Zoals. Patch op een entity. 
![test partially update](https://github.com/user-attachments/assets/14c56e31-0b5d-437a-a8de-326d9c6d45e0)
Daarnaast heb je gefixt de tests.
![test partially update 1](https://github.com/user-attachments/assets/a769eec2-f24d-4ac5-a01d-ea71eb305422)
Hiernaast had ik meerdere updated getest en voor je hieronder gemarked.
1.Name = wel (in clients met id 1)
2.Reference= wel(in orders met id 1)
3.Item_Reference = niet werkt (inventories met id 9273)
![test partially update 2 png ](https://github.com/user-attachments/assets/51c3d23c-16ff-4440-a307-65173e4cf7f5)
4. "Name" = werkt niet (item_types met id 1)
![test partially update 3](https://github.com/user-attachments/assets/c658a044-3309-4daf-96c0-bf510e886e2f)

Sin0fGREED committed:
Patch method opgelost Met hulp van Yegor

YES!

Sin0fGREED committed:
Update inventories.cs

linkverk commented:
Goed gedaan Pratik, je hebt gefixt errors. Je code werkt nu. De locations in inventory ook kan updaten. De Update_At goed werkt, het update naar de tijd van updaten, en niet met data die wordt ingevoerd.
